### ISSUE TITLE
Feature request: passing data to stdin for build commands

### ISSUE THREAD

**Issue by mscdex** [geany/geany#1013]
It would be nice to be able to pass the current document, selected text, etc. to the stdin of a build command. This would be particularly useful for cases where you don't have a new file saved yet, but still want to lint the document (or part of it) for example.

**Comment by b4n** [comment]
What about a [custom command](https://geany.org/manual/index.html#sending-text-through-custom-commands) for that purpose?

**Comment by elextr** [comment]
On 29 April 2016 at 06:02, Colomban Wendling <notifications@github.com>
wrote:

> What about a custom command
> <https://geany.org/manual/index.html#sending-text-through-custom-commands>
> for that purpose?
>

​Is it safe to replace the file with the lint output?​

**Comment by b4n** [comment]
Good point, it's not, yet one wants to see the output.  So nevermind that suggestion, it wasn't worth anything.

**Comment by elextr** [comment]
This doesn't seem likely to have a significant use-case.  In general lint should not be run without previously compiling the source successfully.  An erroneous source file is likely to cause lint to emit many false messages.  But generally compiling requires a file, since compilers use the extension to decide what to do.

**Comment by mscdex** [comment]
@elextr I had something like JavaScript in mind actually

**Comment by elextr** [comment]
Well I guess "somebody" will have to design how to configure build commands to send the file to stdin, and also actually implement the functionality.

**Comment by b4n** [comment]
No spoiler ahead, but a little time ago I was thinking of writing a all-purpose command spawner with all the missing features we currently have, including arbitrary placeholders, environment modifications, selectable stdin/stdout/stderr, etc.  It should actually be fairly easy to implement the feature itself with the new spawn API we have, and it'd be a "mere matter" of writing a nice configurable UI for that.  If I get time I'll try to work on that further.

**Comment by elextr** [comment]
@b4n don't forget to parse the output for file:line so its possible to jump to it by clicking on the output

**Comment by b4n** [comment]
a neat design for that would require being able to customize the available action, so have easy scriptability… we'll see.
